### PR TITLE
DCOS-13362: [7/7] Error Normalisation: Unanchored message support

### PR DESCRIPTION
**Pull request by wavesoft**
---
⚠️  This PR depends on #1839 - ( [See the diff](https://github.com/dcos/dcos-ui/compare/ic/feat/error-message-refactoring-6...ic/feat/error-message-refactoring-7?expand=1) )

---

This PR introduces ErrorMessageUtil.getUnanchoredErrorMessage that prepends location to error messages. In order to provide a user-friendly message we are also using the ServiceErrorPathMapping table that maps paths to descriptive messages.

_This PR implements the `Error Path Translation` task in the spec document: https://docs.google.com/document/d/1kwdPxBo3UlRZvMxLLIUqSZqLlUJitEuIAWderUgWY1c/edit#heading=h.thdpdh4mx6uj_

### CODE REVIEW

File: src/js/utils/ErrorMessageUtil.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   * @param {Array} pathTranslationRules - The path translation rules
+   * @returns {String} Returns the composed error string
+   */
+  getUnanchoredErrorMessage(error, pathTranslationRules) {
+    const pathString = error.path.join('.');
+    const rule = pathTranslationRules.find(function (rule) {
+      return rule.match.exec(pathString);
+    });
+
+    // If no translation found, prefix with the path ID
```
Comment (nLight) [inline src/js/utils/ErrorMessageUtil.js]: This comment looks like it describes line 25 so `return \`${pathString}: ${error.message}\``

File: src/js/utils/ErrorMessageUtil.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if (!rule && !pathString) {
+      return error.message;
+    }
+    if (!rule) {
+      return `${pathString}: ${error.message}`;
+    }
+
+    // Otherwise prefix with the name from the rule
```
Comment (nLight) [inline src/js/utils/ErrorMessageUtil.js]: In comments I'd like to see not `what` but rather `why`.
Maybe I'm out of the context but I don't know why do we want to prefix or lowercase something.
Maybe a small chat would help

File: src/js/utils/ErrorMessageUtil.js (hunk 1 above)
Comment (wavesoft) [inline src/js/utils/ErrorMessageUtil.js]: Good point. Fixed 👍

File: src/js/utils/ErrorMessageUtil.js (hunk 2 above)
Comment (wavesoft) [inline src/js/utils/ErrorMessageUtil.js]: Updated comment 👍

**Comment by wavesoft** [comment]
Rebased 👍

File: src/js/utils/ErrorMessageUtil.js [hunk 3]
```diff
@@ -3,6 +3,36 @@ const MESSAGE_VARIABLE = /\{\{([^{}]+)\}\}/g;
...
   /**
+   * This function returns the `message` of the given array object,
+   * with the path and message components concatenated into a uniform
+   * string that can be displayed unanchored.
+   *
+   * @param {Object} error - The error message
+   * @param {Array} pathTranslationRules - The path translation rules
+   * @returns {String} Returns the composed error string
+   */
+  getUnanchoredErrorMessage(error, pathTranslationRules) {
```
Comment (orlandohohmeier) [inline src/js/utils/ErrorMessageUtil.js]: I'm not sure about the name... How about splitting this into a `translateErrorPath` and a concat util? #nit

Comment (orlandohohmeier) [inline src/js/utils/ErrorMessageUtil.js]: Discussed the topic with @wavesoft offline. We're going to keep it as is until there is a use-case where we need translate/concat separately.

**Comment by orlandohohmeier** [comment]
run unit tests

**Pull request merged**